### PR TITLE
Add events and checks to create_auction + tests

### PR DESCRIPTION
**Pull request by THenry14**

### CODE REVIEW

File: src/main.cairo [hunk 1]
```diff
@@ -180,6 +192,54 @@ func prolong_auction_on_end{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, ra
     return ()
 end
 
+func verify_auction_does_not_exist{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(auction_id):
```
Comment (Solpatium) [inline src/main.cairo]: Rename all verify functions to assert and move to file assertions

Comment (Solpatium) [reply src/main.cairo]: State will have to be moved to a different file for this reason, but this will simplify the code

File: src/main.cairo [hunk 2]
```diff
@@ -180,6 +192,54 @@ func prolong_auction_on_end{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, ra
...
+        let (value) = uint256_lt(Uint256(1, 0), min_bid_increment)
+        assert value = 1
+    end
+
+    return ()
+end
+
+func verify_lifetime{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(lifetime):
+    with_attr error_message(
+        "Invalid lifetime, has to be >= 0"
```
Comment (Solpatium) [inline src/main.cairo]: `>`

Comment (THenry14) [reply src/main.cairo:234]: If lifetime is 0, doesn't that mean the auction will end in the same block it was created? Should we allow for that?

**Pull request merged**